### DISCUSSION TITLE
why get different Variables?

### DISCUSSION THREAD

**Discussion by Afzal1547** [yalmip/YALMIP#1365] [General discussion]
Hello Sir, I am new here, using YALMIP solver in Matlab. Here i don't understant. its just a simple eg.
a = sdpvar(3,3);
b = [1 0 0;0 0 0;1 1 1];

c = a * b
d = a .* b
The result is showing that 
Linear matrix variable 3x3 (full, real, 5 variables)
Linear matrix variable 3x3 (full, real, 4 variables)

but here 'a' is Linear matrix variable 3x3 (symmetric, real, 6 variables).
my Question is why and how we get different variables for 'c' and 'd'? Thanks

**Comment by johanlofberg** [comment]
Because multiplication and elementwise multiplication are two different things

```
>> sdisplay(c)

ans =

  3×3 cell array

    {'a(1,1)+a(3,1)'}    {'a(3,1)'}    {'a(3,1)'}
    {'a(2,1)+a(3,2)'}    {'a(3,2)'}    {'a(3,2)'}
    {'a(3,1)+a(3,3)'}    {'a(3,3)'}    {'a(3,3)'}

>> sdisplay(d)

ans =

  3×3 cell array

    {'a(1,1)'}    {'0'     }    {'0'     }
    {'0'     }    {'0'     }    {'0'     }
    {'a(3,1)'}    {'a(3,2)'}    {'a(3,3)'}

>> 
```
